frontend/www/js/omegaup/components/identity/Editv2.vue

```diff
@@ -0,0 +1,172 @@
+                <option value="other">{{ T.wordsGenderOther }}</option>
+                <option value="decline">{{ T.wordsGenderDecline }}</option>
+              </select>
+            </label>
+          </div>
+          <div class="form-group col-lg-4 col-md-6 col-sm-6">
+            <label class="d-block">
+              {{ T.userEditCountry }}
+              <select v-model="selectedCountry" class="form-control">
+                <option
+                  v-for="country in countries"
+                  :key="country.country_id"
+                  :value="country.country_id"
+                >
+                  {{ country.name }}
+                </option>
+              </select>
+            </label>
+          </div>
+          <div class="form-group col-lg-4 col-md-6 col-sm-6">
+            <label class="d-block">
+              {{ T.profileState }}
+              <select v-model="selectedState" class="form-control">
+                <option
+                  v-for="[code, state] in Object.entries(countryStates)"
+                  :key="code"
+                  :value="code.split('-')[1]"
+                >
+                  {{ state.name }}
+                </option>
+              </select>
+            </label>
+          </div>
+          <div class="form-group col-lg-4 col-md-6 col-sm-6">
+            <label class="d-block">
+              {{ T.profileSchool }}
+              <omegaup-autocomplete
+                v-model="school"
+                class="form-control"
+                :init="(el) => typeahead.schoolTypeahead(el)"
+              ></omegaup-autocomplete>
+              <input type="hidden" :value="schoolId" />
+            </label>
+          </div>
+        </div>
+        <div class="form-group float-right">
+          <button class="btn btn-primary">{{ T.wordsSaveChanges }}</button>
+          <button
+            class="btn btn-secondary ml-2"
+            type="reset"
+            @click="$emit('cancel')"
+          >
+            {{ T.wordsCancel }}
+          </button>
+        </div>
+      </form>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Vue, Component, Watch, Prop } from 'vue-property-decorator';
+import { omegaup } from '../../omegaup';
+import T from '../../lang';
+import * as iso3166 from '@/third_party/js/iso-3166-2.js/iso3166.min.js';
+import * as typeahead from '../../typeahead';
+import Autocomplete from '../Autocomplete.vue';
+
+@Component({
+  components: {
+    'omegaup-autocomplete': Autocomplete,
+  },
+})
+export default class IdentityEdit extends Vue {
+  @Prop() identity!: omegaup.Identity;
+  @Prop() countries!: iso3166.Country[];
+
+  T = T;
+  typeahead = typeahead;
+  username = this.identity.username;
+  name = this.identity.name;
+  gender = this.identity.gender;
+  school = this.identity.school;
+  schoolId = this.identity.school_id;
+  selectedCountry = this.identity.country_id ?? 'MX';
+  selectedState = this.identity.state_id;
```

Comment:
me pregunto si es mejor declarar un

```suggestion
  selectedIdentity = Object.assign(<omegaup.Identity>{name: '', gender: ...}, this.identity);
```

así no es necesario desconstruir el objeto y L121-L127 pueden ser `Object.assign(this.identity, newIdentity);`.

Reply:
A ver si así está bien.